vpr/src/place/place_constraints.cpp

```diff
@@ -32,11 +33,83 @@ bool is_cluster_constrained(ClusterBlockId blk_id) {
     auto& floorplanning_ctx = g_vpr_ctx.floorplanning();
     PartitionRegion pr;
     pr = floorplanning_ctx.cluster_constraints[blk_id];
-    if (pr.empty()) {
-        return false;
-    } else {
-        return true;
+    return (!pr.empty());
+}
+
+bool is_macro_constrained(t_pl_macro pl_macro) {
+    bool is_macro_constrained = false;
+    bool is_member_constrained = false;
+
+    for (size_t imember = 0; imember < pl_macro.members.size(); imember++) {
```

Comment:
see if you need this function. have a new version of is_macro_constrained that gets rid of the for loop

Reply:
in general, go through your code and make sure you are passing by reference instead of value when you can